puffin_egui/src/stats.rs

```diff
@@ -8,11 +8,98 @@ pub struct Options {
     filter: Filter,
 }
 
+#[derive(Copy, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+pub enum SortKey {
+    Location,
+    FunctionName,
+    ScopeName,
+    Count,
+    Size,
+    TotalSelfTime,
+    MeanSelfTime,
+    MaxSelfTime,
+}
+
+#[derive(Copy, Clone)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+pub struct SortOrder {
+    pub key: SortKey,
+    pub rev: bool,
+}
+
+impl SortOrder {
+    fn sort_scopes(&self, scopes: &mut [(&Key, ScopeStats)]) {
+        match self.key {
+            SortKey::Location => {}
+            SortKey::FunctionName => {}
+            SortKey::ScopeName => {
+                /*
+                scopes.sort_by_key(|(_key, scope_stats)| {
+                    scope_infos.fetch_by_id(&key.id).unwrap().location()
+                });
+                */
+            }
+            SortKey::Count => {
+                scopes.sort_by_key(|(_key, scope_stats)| scope_stats.count);
+            }
+            SortKey::Size => {
+                scopes.sort_by_key(|(_key, scope_stats)| scope_stats.bytes);
+            }
+            SortKey::TotalSelfTime => {
+                scopes.sort_by_key(|(_key, scope_stats)| scope_stats.total_self_ns);
+            }
+            SortKey::MeanSelfTime => {
+                scopes.sort_by_key(|(_key, scope_stats)| {
+                    scope_stats.total_self_ns as usize / scope_stats.count
+                });
+            }
+            SortKey::MaxSelfTime => {
+                scopes.sort_by_key(|(_key, scope_stats)| scope_stats.max_ns);
+            }
+        }
+        if self.rev {
+            scopes.reverse();
+        }
+    }
+
+    fn get_arrow(&self) -> &str {
+        if self.rev {
+            "v"
+        } else {
+            "^"
```

Comment:
```suggestion
            "⏶"
```

Reply:
See also https://docs.rs/egui/latest/egui/special_emojis/index.html and the egui font book at egui.rs